### DISCUSSION TITLE
How do I use the "Item Modifier Generator" ?

### DISCUSSION THREAD

**Discussion by DarkFluo** [misode/misode.github.io#406] [Q&A]
Hey, quick question, a pretty stupid one I think, but I haven't found any documentation on the website or GitHub. I imagine this is all based on common DataPack knowledge, but to someone like me that doesn't know much about all that it's pretty confusing.

I'm trying out this website to make some small datapacks for a server I wanna create with my friends, and I'm having trouble making anything work. For now I'm simply trying to have a pickaxe get enchanted to Efficiency 1 as soon as the player has it in it's hand. I don't know if it's possible using this website, but the Item Modifier Tool seems to have a similar purpose ?

I tried this, but it didn't work :

```
{
  "function": "minecraft:set_enchantments",
  "enchantments": {
    "minecraft:efficiency": 1
  },
  "conditions": [
    {
      "condition": "minecraft:entity_properties",
      "entity": "this",
      "predicate": {
        "type": "minecraft:player",
        "equipment": {
          "mainhand": {
            "items": [
              "minecraft:diamond_pickaxe"
            ],
            "enchantments": [
              {
                "enchantment": "minecraft:efficiency",
                "levels": 0
              }
            ]
          }
        }
      }
    }
  ]
}
```

If there are any documentation I can use for simple instructions, please point them out, I don't think y'all would be very happy about some guy posting 500 posts asking basic datapack related things...

I hope I'm not asking for something too simple it could be looked up by myself, I haven't found anything while searching for this precise purpose and I don't really want to get a PHD in DataPacks creation just to make a couple of simple packs for a month long server. This website seemed like an easy way to get these basic things going, but maybe I'm wrong ? I hope you'll be able to help me on that one.

**Comment by Ethanout** [comment]
```
execute as @a if items entity @s weapon.mainhand diamond_pickaxe run item modify entity @s weapon.mainhand {"function":"minecraft:set_enchantments","enchantments":{"minecraft:efficiency":1},"add":false}
```
